### ISSUE TITLE
Date utils Month subtraction fails on the 31st of Jan 2022

### ISSUE THREAD

**Issue by AlphaJuliettOmega** [quasarframework/quasar#12270]
### What happened?

```
import { date } from 'quasar'
      const monthsToBeRetrieved = 3
      const today = Date.now()
      for (let i = this.monthsBack; i < this.monthsBack + monthsToBeRetrieved; i++) {
        const month = date.subtractFromDate(today, { months: i })
        const formatMonth = date.formatDate(month, 'YYYY-MM')
```

Today, this loop creates the dates January, December AND DECEMBER
As if subtracting a month subtracts 30 days, instead of the calendar month length.

### What did you expect to happen?

I expected the date.subtractFromDate to subtract a month accurately, as Momentjs does it because that's what's said in the documentation:

> ... and remember that you don’t need solutions like Momentjs which add hundreds of minified KB to your bundle.

### Reproduction URL

...

### How to reproduce?

Copy paste For loop above, run code on 31 January or use Timekeeper to set date.

### Flavour

Quasar CLI (@quasar/cli | @quasar/app)

### Areas

Quasar CLI Commands/Configuration (@quasar/cli | @quasar/app), Plugins (quasar), Quasar Extras (@quasar/extras)

### Platforms/Browsers

Firefox, Chrome, Safari, Microsoft Edge, iOS, Android, Electron, Other

### Quasar info output

```shell
Operating System - Darwin(21.2.0) - darwin/x64
NodeJs - 14.18.2

Global packages
  NPM - 6.14.15
  yarn - 1.22.17
  @quasar/cli - 1.2.2
  @quasar/icongenie - Not installed
  cordova - Not installed

Important local packages
  quasar - 2.4.12 -- Build high-performance VueJS user interfaces (SPA, PWA, SSR, Mobile and Desktop) in record time
  @quasar/app - 3.2.9 -- Quasar Framework local CLI
  @quasar/extras - 1.12.4 -- Quasar Framework fonts, icons and animations
  eslint-plugin-quasar - Not installed
  vue - 3.2.24 -- The progressive JavaScript framework for buiding modern web UI.
  vue-router - 4.0.12
  vuex - 4.0.2 -- state management for Vue.js
  electron - Not installed
  electron-packager - Not installed
  electron-builder - Not installed
  @babel/core - 7.16.0 -- Babel compiler core.
  webpack - 5.65.0 -- Packs CommonJs/AMD modules for the browser. Allows to split your codebase into multiple bundles, which can be loaded on demand. Support loaders to preprocess files, i.e. json, jsx, es7, css, less, ... and your custom stuff.
  webpack-dev-server - 4.7.3 -- Serves a webpack app. Updates the browser on changes.
  workbox-webpack-plugin - Not installed
  register-service-worker - 1.7.2 -- Script for registering service worker, with hooks
  typescript - 4.5.4 -- TypeScript is a language for application scale JavaScript development
  @capacitor/core - Not installed
  @capacitor/cli - Not installed
  @capacitor/android - Not installed
  @capacitor/ios - Not installed

Quasar App Extensions
  @quasar/quasar-app-extension-testing - 2.0.2 -- A Quasar App Extension for managing Test Harnesses
  @quasar/quasar-app-extension-testing-unit-jest - 3.0.0-alpha.7 -- A Quasar App Extension for running Jest tests
```

### Relevant log output

```shell
...
month 0 Mon Jan 31 2022 10:43:52 GMT+0200 (South Africa Standard Time)
...?394a:78 month 1 Fri Dec 31 2021 10:43:52 GMT+0200 (South Africa Standard Time)
...?394a:78 month 2 Wed Dec 01 2021 10:43:52 GMT+0200 (South Africa Standard Time)
```

**Comment by AlphaJuliettOmega** [comment]
A reproduction URL is not needed for this issue, the code is clear

**Comment by AlphaJuliettOmega** [comment]
Related perhaps (but closed):
https://github.com/quasarframework/quasar/issues/9704

Also related:
https://github.com/quasarframework/quasar/issues/12032

**Comment by yusufkandemir** [comment]
There is no such thing as repro URL is not needed. Also, the code is not very clear considering we don't know the value of `this.monthsBack`, but with a guess, it must be 0.

Here is a small and clear reproduction of the issue:
https://codepen.io/yusufkandemir/pen/yLPeJyr?editors=1010

**Comment by rstoenescu** [comment]
Fix will be available in Quasar v2.5.0
